packages/oft-evm/contracts/OFTFee.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import { ERC20, IERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { IFee } from "./interfaces/IFee.sol";
+import { OFTCore } from "./OFTCore.sol";
+import { OFT } from "./OFT.sol";
+import { Fee } from "./Fee.sol";
+
+/**
+ * @title OFTFee Contract
+ * @dev OFT is an ERC-20 token that extends the functionality of the OFTCore contract.
+ */
+abstract contract OFTFee is OFT, Fee {
+    /**
+     * @dev Constructor for the OFT contract.
+     * @param _name The name of the OFT.
+     * @param _symbol The symbol of the OFT.
+     * @param _lzEndpoint The LayerZero endpoint address.
+     * @param _delegate The delegate capable of making OApp configurations inside of the endpoint.
+     */
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        address _lzEndpoint,
+        address _delegate
+    ) OFT(_name, _symbol, _lzEndpoint, _delegate) {}
+
+    /**
+     * @notice Indicates whether the OFT contract requires approval of the 'token()' to send.
+     * @return requiresApproval Needs approval of the underlying token implementation.
+     *
+     * @dev In the case of OFT where the contract IS the token, approval is NOT required.
+     */
+    function approvalRequired() external pure virtual override returns (bool) {
+        return true;
+    }
+
+    /**
+     * @dev Burns tokens from the sender's specified balance.
+     * @param _from The address to debit the tokens from.
+     * @param _amountLD The amount of tokens to send in local decimals.
+     * @param _minAmountLD The minimum amount to send in local decimals.
+     * @param _dstEid The destination chain ID.
+     * @return amountSentLD The amount sent in local decimals.
+     * @return amountReceivedLD The amount received in local decimals on the remote.
+     */
+    function _debit(
+        address _from,
+        uint256 _amountLD,
+        uint256 _minAmountLD,
+        uint32 _dstEid
+    ) internal virtual override returns (uint256 amountSentLD, uint256 amountReceivedLD) {
+        (amountSentLD, amountReceivedLD) = _debitView(_amountLD, _minAmountLD, _dstEid);
+        _transferFrom(_from, address(this), amountSentLD - amountReceivedLD);
```

Comment:
This should be a setter, which defaults to address(this), but can be overriden by the owner to send direct to their fee colelctor

Reply:
I disagree;  let me explain my thinking.  This implementation would lead to two different usage patterns for distributing fees:

1) Collect and lock fees in `address(this)`.  In the case of `OFTAdapter`, this obviously comingles fees and locked assets.  In this case `OFTFeeAdapter.feeBalance` is used to tally fees.  Later, the owner may `withdrawFees()`.
2) Provide a hook to set a receiver, and immediately distribute fees to that receiver.

I would argue for just the former.  My reason is that, especially with the case of AdapterOFT, funds are just moved once so it is cheaper (i.e., only locked in the OFTFeeAdapter contract), and fees are distinguished from locked assets using the `feeBalance`.  For the latter, we would need to implement complicated logic to check who the receiver is, and only add fees to `OFTFeeAdapter.feeBalance` if the receiver is not the contract itself (i.e., the funds never accrue because they are always immediately distributed).